src/lib.rs

```diff
@@ -699,8 +712,35 @@ pub struct DefaultHandler {
 }
 
 impl DefaultHandler {
+    /// Manual hook which constructs `DefaultHandler`s.
+    ///
+    /// # Details
+    ///
+    /// When supplied to the `set_hook` function, `default_with` will cause `eyre::Report` to use
+    /// `DefaultHandler` as the error report handler.
+    ///
+    /// If the `auto-install` feature is enabled, and a user-provided hook for constructing
+    /// `EyreHandlers` was not installed using `set_hook`, `DefaultHandler::default_with`
+    /// is automatically installed as the hook.
+    ///
+    /// # Example
+    ///
+    /// ```rust,should_panic
+    /// use eyre::{DefaultHandler, eyre, InstallError, Result, set_hook};
+    ///
+    /// fn main() -> Result<()> {
+    ///     install_default().expect("default handler inexplicably already installed");
+    ///     Err(eyre!("hello from default error city!"))
+    /// }
+    ///
+    /// fn install_default() -> Result<(), InstallError> {
+    ///     set_hook(Box::new(DefaultHandler::default_with))
+    /// }
+    ///
+    /// ```
     #[allow(unused_variables)]
-    fn default_with(error: &(dyn StdError + 'static)) -> Box<dyn EyreHandler> {
+    #[cfg_attr(not(feature = "auto-install"), allow(dead_code))]
```

Comment:
does the `.text` size reduction only work if this is cfg'ed away? I would hope we could just never use it and it would get rmoved even if it's pub and still exists.

Assuming that works, I think we need to remove this cfg. This means enabling `auto-install` removes the ability to install a handler manually. I think this would break if someone who disabled the auto-install feature and installed it manually linked in a library that enabled the `auto-install` feature.

Reply:
That `cfg` is _intended_ to just to disable the `allow(dead_code)` lint, not remove `default_with` if `auto-install` is enabled.

Reply:
oh duh, ty